File: EloHeaven.Services.Logic/Regions/RegionService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EloHeaven.Data;
using EloHeaven.Entities;
using EloHeaven.Logic;

namespace EloHeaven.Services.Logic.Regions
{
    public class RegionService : IRegionService
    {
        private readonly IRegionRepository _regionRepository;
        private readonly IModelMapper<Region, RegionModel> _regionModelMapper; 

        public RegionService(IRegionRepository regionRepository, IModelMapper<Region, RegionModel> regionModelMapper)
        {
            _regionRepository = regionRepository;
            _regionModelMapper = regionModelMapper;
        }

        public IEnumerable<RegionModel> GetAll()
        {
            return _regionRepository.GetAll().Select(_regionModelMapper.ToModel).ToList();
        }
    }
}

File: EloHeaven/App/region.service.js
﻿(function() {

    angular
        .module("eloHeaven")
        .factory("regionService", regionService);

    function regionService($http) {
        var service = {
            getAll: getAll
        };

        return service;

        function getAll() {
            return $http.get("/api/region").then(function(response) {
                return response.data;
            });
        }
    }

})();